matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/relation/threads/FetchThreadTimelineTask.kt

```diff
@@ -69,93 +92,132 @@ internal class DefaultFetchThreadTimelineTask @Inject constructor(
+                            direction = PaginationDirection.FORWARDS,
+                            ownedByThreadChunk = true,
+                            roomMemberContentsByUser = roomMemberContentsByUser)
                 }
-        Timber.i("----> size: ${threadList.size} | skipped: $eventsSkipped | threads: ${threadList.map { it.eventId }}")
+            }
 
-        return eventsSkipped == threadList.size
+            if (hasReachEnd) {
+                val rootThread = TimelineEventEntity
+                        .where(realm, roomId = params.roomId, params.rootThreadEventId)
+                        .findFirst()
+                if (rootThread != null) {
+                    // If root thread event already exists add it to our chunk
+                    threadChunk.timelineEvents.add(rootThread)
+                    Timber.i("###THREADS FetchThreadTimelineTask root thread event: ${params.rootThreadEventId} found and added!")
+                } else if (threadRootEvent?.senderId != null) {
+                    // Case when thread event is not in the device
+                    Timber.i("###THREADS FetchThreadTimelineTask root thread event: ${params.rootThreadEventId} NOT FOUND! Lets create a temp one")
+                    val eventEntity = createEventEntity(params.roomId, threadRootEvent, realm)
+                    roomMemberContentsByUser.addSenderState(realm, params.roomId, threadRootEvent.senderId)
+                    threadChunk.addTimelineEvent(
+                            roomId = params.roomId,
+                            eventEntity = eventEntity,
+                            direction = PaginationDirection.FORWARDS,
+                            ownedByThreadChunk = true,
+                            roomMemberContentsByUser = roomMemberContentsByUser)
+                }
+            }
+        }
+
+        return if (hasReachEnd) {
+            Result.REACHED_END
+        } else {
+            Result.SHOULD_FETCH_MORE
+        }
     }
 
+    // TODO Reuse this function to all the app
     /**
-     * Invoke the event decryption mechanism for a specific event
+     * If we don't have any new state on this user, get it from db
      */
+    private fun HashMap<String, RoomMemberContent?>.addSenderState(realm: Realm, roomId: String, senderId: String) {
+        getOrPut(senderId) {
+            CurrentStateEventEntity
+                    .getOrNull(realm, roomId, senderId, EventType.STATE_ROOM_MEMBER)
+                    ?.root?.asDomain()
+                    ?.getFixedRoomMemberContent()
+        }
+    }
 
+    /**
+     * Create an EventEntity to be added in the TimelineEventEntity
+     */
+    private fun createEventEntity(roomId: String, event: Event, realm: Realm): EventEntity {
+        val ageLocalTs = event.unsignedData?.age?.let { System.currentTimeMillis() - it }
+        return event.toEntity(roomId, SendState.SYNCED, ageLocalTs).copyToRealmOrIgnore(realm, EventInsertType.INCREMENTAL_SYNC)
+    }
```

Comment:
The idea is to mimic the same behaviour with the incremental sync, Thread Timeline now is **temporarily** and will be soft deleted when the user exits or enters again the thread timeline (by soft delete I mean that `TimelineEvent` is deleted but `Event` will remain so it will be reused when the original timeline fetch that event again). So keeping it as `INCREMENTAL_SYNC` will make the app continue having the same behaviour, I cannot see a specific problem with that but if you can see something we can discuss.

Reply:
Wouldn't be better to use pagination ? it's for some internal processing

Reply:
Hmm, maybe that will work too. I will check this out and update it accordingly, thanks.

Reply:
Updated accordingly, seems there is no issue with that change